### ISSUE TITLE
Access a specific item in a subfield?

### ISSUE THREAD

**Issue by AWegnerGitHub** [pawelzny/dotty_dict#37]
I have an example `dict` that looks like this:

```
{
        'field1': 'Value of F1',
        'field2': 'Value of F2',
        'field3': [
                {
                    'subfield1': 'Value of subfield1 (item 0)',
                    'subfield2': 'Value of subfield2 (item 0)'
                },
                {
                    'subfield1': 'Value of subfield1 (item 1)',
                    'subfield2': 'Value of subfield2 (item 1)'
                },
            ],
        'field4': 'Not wanted'
    }
```

---

This is my full script to test:

```
from dotty_dict import dotty
wanted_fields = ['field1', 'field2', 'field3.subfield1']
# field3.subfield1 works only if field3 isn't a list, but a single item
# field3.0.subfield1 throws a type error
# field3[0].subfield1 throws a keyerror

devices = [
    {
        'field1': 'Value of F1',
        'field2': 'Value of F2',
        'field3': [
                {
                    'subfield1': 'Value of subfield1',
                    'subfield2': 'Value of subfield2'
                }
            ],
        'field4': 'Not wanted'
    }
]

for device in devices:
    devices_map = dotty(device)
    for item in wanted_fields:
        print("{item} => {value}".format(item=item, value=devices_map[item]))

```

---

How can I get just element 0 from the `field3` key?

I tried `field3.0.subfield1` and `field3[0].subfield1`. Neither works. 

With the first, I receive a TypeError: `TypeError: list indices must be integers or slices, not str`
With the second, I receive a KeyError: `KeyError: 'field3[0]'`

Additionally, how can I get *every* `subfield1` in the `field3` list?

**Comment by pawelzny** [comment]
Right now lists are not supported. Dotty recursively checks dicts values and if value is another dict it will convert it to dotty on the fly otherwise it stops. This is why you get TypeError because it is no longer the dotty object but a list.

In your example it is possible to get first element of the list, but this element must be converted to dotty manually.

```python
devices_map = dotty(device)
device_field = dotty(devices_map['field3'][0])
```

I think it is a good idea to support lists in form of `field.0.subfield`. I will do my best to implement it before Easter.

**Comment by pawelzny** [comment]
v.1.1.0 has been released. 
It supports accessing subfields through list index, update subfield values and delete subfields.

It is already available on PyPI `pip install dotty_dict==1.1.0`

For drop in replacement of your current code please use alias:

```python
from dotty_dict import dotty_l as dotty
```
